src/lib.rs

```diff
@@ -200,8 +196,9 @@ impl Pty {
     thread::spawn(move || {
       let wait_result = child.wait();
 
-      // try to wait for the controller fd to be fully read 
-      poll_controller_fd_until_read(raw_controller_fd);
+      // try to wait for the controller fd to be fully read
+      let controller_fd = unsafe { OwnedFd::from_raw_fd(raw_controller_fd) };
+      poll_controller_fd_until_read(controller_fd);
```

Comment:
hmm i think this is unsafe, because we don't truly own the raw controller fd in this context, and exiting the poll will `.close` it! but then we will `.close` it again in `.close()`!!!

we probably need to reintroduce the fd copying that we had earlier :sob:.